src/ucs/datastruct/bitmap.h

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_BITMAP_H_
+#define UCS_BITMAP_H_
+
+#include <ucs/sys/compiler_def.h>
+
+BEGIN_C_DECLS
+
+#define _UCS_BITMAP_BITS_IN_WORD (64)
+#define _UCS_BITMAP_WORD_INDEX(_bit_index) ((_bit_index) / _UCS_BITMAP_BITS_IN_WORD )
+#define _UCS_BITMAP_BIT_IN_WORD_INDEX(_bit_index) ((_bit_index) % _UCS_BITMAP_BITS_IN_WORD)
+
+/*
+ * Represents an n-bit bitmap, where n is a multiple of 64, by using an array
+ * of 64-bit unsigned long integers. Memory is allocated on stack
+ * 
+ * @param _bits Bits number in the bitmap
+ */
+#define _UCS_BITMAP_DECLARE_TYPE(_bits) \
+    typedef struct { \
+        uint64_t bits[_UCS_BITMAP_WORD_INDEX(_bits)]; \
```

Comment:
plz document that it works for power of 2 number of bits only

Reply:
btw, if use 1-byte based array then any size of bitmap may be used

Reply:
@brminich - It is already documented (lines 18-23).
@hoopoepg - yes, but would that improve anything?